### DISCUSSION TITLE
Read only for existing annotations?

### DISCUSSION THREAD

**Discussion by arlendarcy** [mozilla/pdf.js#18782] [Q&A]
I am looking for a way to disable the editing of existing (annotations that already exist on the document upon load) annotations while still retaining the ability to add new ones. 

Looking through the code that handles annotations it appears that without access to `AnnotationStorage.#storage` this will not be possible, but I was hoping I just missed something or am not thinking about this in the right way. 

Any advice would be greatly appreciated!

**Comment by marco-c** [comment]
@arlendarcy what is your use case?

**Reply by arlendarcy** [reply]
Thanks for your reply @marco-c. 

Displaying prescription faxes that have annotations on them. Any saved annotations should not be editable for auditability. Annotations could have already been added to the faxes before our application receives them. Does that help?

**Reply by arlendarcy** [reply]
Any thoughts on this matter?
